arangod/RestHandler/RestReplicationHandler.cpp

```diff
@@ -2700,16 +2589,144 @@ ReplicationApplier* RestReplicationHandler::getApplier(bool& global) {
   }
 }
 
-//////////////////////////////////////////////////////////////////////////////
-/// @brief condition locker to wake up holdReadLockCollection jobs
-//////////////////////////////////////////////////////////////////////////////
+Result RestReplicationHandler::createBlockingTransaction(aql::QueryId id,
+                                                         LogicalCollection& col,
+                                                         double ttl) const {
+  // This is a constant JSON structure for Queries.
+  // We do not need nodes or variables.
+  // We only need collections, with corresponding lock type
+  auto planBuilder = std::make_shared<VPackBuilder>();
+  planBuilder->openObject();
+  planBuilder->close();
+
+  auto query = std::make_unique<aql::Query>(
+    false,
+    _vocbase,
+    planBuilder,
+    nullptr, /* options */
+    aql::QueryPart::PART_MAIN /* Do locking */
+  );
+ // NOTE: The collections are on purpose not locked here.
+  // To accuire an EXCLUSIVE lock may require time under load,
+  // we want to allow to cancel this operation while waiting
+  // for the lock.
 
-arangodb::basics::ConditionVariable RestReplicationHandler::_condVar;
+  auto queryRegistry = QueryRegistryFeature::QUERY_REGISTRY;
+  TRI_ASSERT(queryRegistry != nullptr);
```

Comment:
can we remove these asserts if we cannot guarantee that there will be no shutdown?

Reply:
I think we have to delete the assert if the code below makes any sense, which I believe it does.